test/stack_controller_test.go

```diff
@@ -216,7 +216,7 @@ var _ = Describe("Stack Controller", func() {
 				return false
 			}
 			if fetched.Status.LastUpdate != nil {
-				return fetched.Status.LastUpdate.State == commit
+				return fetched.Status.LastUpdate.LastSuccessfulCommit == commit
```

Comment:
Are there tests we can add that cover state transitions for `lastAttemptedCommit` and `state` as well?

Reply:
Yup - updated now.